parse.y

```diff
@@ -80,6 +80,94 @@ struct lex_context;
 #include "ruby/ractor.h"
 #include "symbol.h"
 
+static inline int
+parse_isspace(int c)
+{
+    return c == ' ' || ('\t' <= c && c <= '\r');
+}
+
+#ifdef ISSPACE
+#undef ISSPACE
+#define ISSPACE parse_isspace
+#endif
```

Comment:
Is `#ifdef` needed?
This means `ISSPACE` macro needs to be defined previously.

Reply:
Thanks for comment.
`ISSPACE` macro already defined in `include/ruby/internal/ctype.h` and that included in `ruby/ruby.h`.
`parse.y` include `ruby/ruby.h`, and I thought `#undef ISSPACE` is needed.

Reply:
At the moment `include/ruby/internal/ctype.h` is included into parse.y in both cases of with `UNIVERSAL_PARSER` and without `UNIVERSAL_PARSER`. However we may remove such inclusion in the future if possible. Current code wrapped with `#ifdef` might not work once `include/ruby/internal/ctype.h` is not included because `#define ISSPACE parse_isspace` is not defined without `ISSPACE` definition ahead of this line.
Could you try to remove `#ifdef ISSPACE` ... `#ifdef STRNCASECMP` ?

Reply:
Thanks comment and I understand. I'll try it.
And shoud I remove also `#ifdef ISASCII `?

Reply:
Removed `#ifdef ISSPACE` ... `#ifdef STRNCASECMP`(also `#ifdef ISASCII`).
It' be fine that all test has passed, I think.

Reply:
Not only `#ifdef ISSPACE` and `#ifdef ISASCII` but also other `#ifdef XXX` like `#ifdef ISCNTRL`, `#ifdef ISALPHA` ...

By the way, current positions for `parse_XXX` is only for without `UNIVERSAL_PARSER`. Could you move these definitions to the place outside of `UNIVERSAL_PARSER` macro branches and `RIPPER` macro branches like https://github.com/ruby/ruby/blob/2903e9456f6990c068008eb823b90715e5a21932/parse.y#L542?

```
# without UNIVERSAL_PARSER
$ nm parse.o | grep isspace
0000000000010ab0 t _parse_isspace
000000000002a9b8 t _rb_enc_isspace

# with UNIVERSAL_PARSER
$ nm parse.o | grep isspace
0000000000010a08 t _rb_isspace
```

Reply:
Thanks comment, It's fixed to move outside of UNIVERSAL_PARSER macro branches and RIPPER macro branche.